webgui-new/InstallReact.cmake

```diff
@@ -0,0 +1,45 @@
+# Utility functions
+include(${CMAKE_SOURCE_DIR}/Functions.cmake)
+
+message("Install npm packages...")
+
+if(${CC_PACKAGE} IS_NEWER_THAN ${INSTALL_WEBROOT_REACT_DIR}/app/package.json)
+  execute_process(
+    COMMAND ${CMAKE_COMMAND} -E copy ${CC_PACKAGE} ${INSTALL_WEBROOT_REACT_DIR}/app/package.json
+    WORKING_DIRECTORY ${INSTALL_WEBROOT_REACT_DIR})
+
+  execute_process(
+    COMMAND npm install
+    WORKING_DIRECTORY ${INSTALL_WEBROOT_REACT_DIR}/app)
+endif()
+
+message("Installation of npm packages are finished.")
+
+# Generate TypeScript from Thrift
+execute_process(
+  COMMAND bash -c "chmod +x thrift-codegen.sh"
+  WORKING_DIRECTORY ${INSTALL_WEBROOT_REACT_DIR}/app)
+
+execute_process(
+  COMMAND bash thrift-codegen.sh --thrift-source ${CMAKE_SOURCE_DIR}
+  WORKING_DIRECTORY ${INSTALL_WEBROOT_REACT_DIR}/app)
+
+message("Building React App...")
+
+execute_process(
+  COMMAND npm run build
+  WORKING_DIRECTORY ${INSTALL_WEBROOT_REACT_DIR}/app)
+
+# Move build directory out of application directory
+file(REMOVE_RECURSE ${INSTALL_WEBROOT_REACT_DIR}/out)
+file(RENAME ${INSTALL_WEBROOT_REACT_DIR}/app/out ${INSTALL_WEBROOT_REACT_DIR}/out)
+
+# Create symbolic link
+execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink ${INSTALL_WEBROOT_REACT_DIR}/out ${INSTALL_WEBROOT_DIR}/new)
```

Comment:
This should use a relative path to be more robust. (E.g. the installation folder can be moved or renamed.)

Reply:
Try building the Docker images 😃 

According to the workflow, first, CodeCompass will be installed into the `/CodeCompass-install` folder during the build stage:
https://github.com/Ericsson/CodeCompass/blob/master/docker/runtime/Dockerfile#L31

Then, it will be moved to the `/codecompass` folder when the runtime image is constructed:
https://github.com/Ericsson/CodeCompass/blob/master/docker/runtime/Dockerfile#L82

Therefore the `new` symlink in the `webgui` folder will be invalid, because it was created using an absolute path. I had to fix that manually in the Docker image, so it works on codecompass.net. Using a relative path for the *symlink* would solve this issue. (Of course in the long run, there will be only one web GUI, but until then.)

Reply:
I see, thanks for clarifying. Altough I'm a little unsure how I would use a relative path here. Could you elaborate on that?

Reply:
Well, instead of `${INSTALL_WEBROOT_REACT_DIR}/out`, it should be simply `../webgui-new/out`.

Reply:
I see, and should `${INSTALL_WEBROOT_DIR}/new` also be changed to `../webgui/new` then?

Reply:
No, `${INSTALL_WEBROOT_DIR}/new` is a correct, valid absolute path, which will only be used when creating the symlink. Only `${INSTALL_WEBROOT_REACT_DIR}/out` should be changed to a relative path, expecting that the `webgui` and the `webgui-new` folder will always be next to each other, so it will work when their parent directory is moved / renamed.